src/smt/set_defaults.cpp

```diff
@@ -311,13 +311,14 @@ void setDefaults(LogicInfo& logic, bool isInternalSubsolver)
     // Note we allow E-matching by default to support combinations of sequences
     // and quantifiers.
   }
-  // whether we must disable proofs
-  bool disableProofs = false;
+  // whether we must disable proofs. There are three levels: no disable, partial
```

Comment:
Why this change?  I don't see why the 2 cases below (global negate and sygus) should be distinguished.

Reply:
I figured it'd be good to discuss this. My motivation was to deal with problems where we use sygus-inference and want unsat cores, basically. Just disabling proofs is not what we wanted. I was not super sure about what to do in the previous cases, so  I kept both modes. Do you see a better way of handling this?

Reply:
`disableProofs` is set to true only when it truly is the case that proofs (and hence unsat cores) do not apply.  This policy should not change in this PR.  Here, `sygus-inference` is always incompatible with unsat cores (actually, as a technique, it only is for "sat" anyways).

Reply:
We do use the option in some unsat regressions, which led to issues when I changed the defaults and motivated the above setup. Should we disable cores for these regressions then?

Reply:
Yep, that does it. Thanks!